packages/now-cli/src/util/env/get-env-records.ts

```diff
@@ -1,18 +1,70 @@
 import { Output } from '../output';
 import Client from '../client';
-import { ProjectEnvVariable, ProjectEnvTarget } from '../../types';
+import {
+  ProjectEnvVariable,
+  ProjectEnvTarget,
+  PaginationOptions,
+} from '../../types';
+import { URLSearchParams } from 'url';
+
+type ApiVersion = 4 | 5;
+
+type APIV4Response = ProjectEnvVariable[];
+
+interface APIV5Response {
+  pagination: PaginationOptions;
+  envs: ProjectEnvVariable[];
+}
+
+export default async function getEnvVariables(
+  output: Output,
+  client: Client,
+  projectId: string,
+  apiVersion: 4,
+  target?: ProjectEnvTarget,
+  next?: number
+): Promise<APIV4Response>;
 
 export default async function getEnvVariables(
   output: Output,
   client: Client,
   projectId: string,
-  target?: ProjectEnvTarget
-): Promise<ProjectEnvVariable[]> {
+  apiVersion: 5,
+  target?: ProjectEnvTarget,
+  next?: number
+): Promise<APIV5Response>;
+
+export default async function getEnvVariables<V extends ApiVersion>(
+  output: Output,
+  client: Client,
+  projectId: string,
+  apiVersion: V,
+  target?: ProjectEnvTarget,
+  next?: number
+) {
   output.debug(
     `Fetching Environment Variables of project ${projectId} and target ${target}`
   );
-  const qs = target ? `?target=${encodeURIComponent(target)}` : '';
-  const url = `/v4/projects/${projectId}/env${qs}`;
-  const records = await client.fetch<ProjectEnvVariable[]>(url);
-  return records;
+  const query = new URLSearchParams();
+  if (apiVersion >= 5) {
+    query.set('limit', String(20));
+  }
+
+  if (target) {
+    query.set('target', encodeURIComponent(target));
```

Comment:
```suggestion
    query.set('target', target);
```

I think this will be encode automatically